### DISCUSSION TITLE
Testing a custom error handler

### DISCUSSION THREAD

**Discussion by allforabit** [testing-library/angular-testing-library#374] [Q&A]
I'm attempting to write a test to ensure a custom error handler is calling a logging service when an error occurs by deliberately mocking a malformed response and thus triggering the error (and the custom error handler). This works in production code but for some reason tests using Angular testing library are throwing an uncaught error and seems to be bypassing the custom error handler. The handler is supplied as a provider similarly to the production config as follows:

```typescript
  return {
    provide: ErrorHandler,
    useClass: ClientErrorHandler,
  };
```

Any help greatly appreciated!

**Comment by timdeschryver** [comment]
Hi, could you please share a full reproduction?

**Reply by allforabit** [reply]
Hi Tim so sorry for not getting back to you sooner as I got a bit sidetracked. I will look at reproducing this over the next few days. Thanks very much.
